### DISCUSSION TITLE
My own certificate

### DISCUSSION THREAD

**Discussion by info100t** [amidaware/tacticalrmm#1424] [General]
Hi team ! :)

**Server Info (please complete the following information):**
 - OS: Debian 10
 - Browser: chrome
 - RMM Version : 0157

**Installation Method:**
  - [X] Standard
  - [ ] Docker

**Good to know**
I use my own certificate.

**The bug**
Agents are overdue.

When i try reinstall agent, meshagent can't download.

When i do " /rmm/api/env/bin/python /rmm/api/tacticalrmm/manage.py check_mesh" i have an error : Error getting mesh device id:
[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)

**Question**

i think it' my cert create the problem. Can you tell me how can i use my own cert ?

thx
romain

**Comment by dinger1986** [comment]
Hello, it’s in the docs :).

https://docs.tacticalrmm.com/unsupported_scripts/#using-purchased-ssl-certs-instead-of-lets-encrypt-wildcards

**Reply by info100t** [reply]
wow... sorry !
i search but i don't find... 
thx
romain

**Reply by dinger1986** [reply]
No worries at all!

**Comment by info100t** [comment]
So... i solve 1 problem with doc ! hum... it was too easy ! :p 

When i install agent, i have a message box "unable to download the mesh agent from the RMM".

romain

**Reply by dinger1986** [reply]
Is the mesh service running now?

Did you run update.eh with force?

**Comment by info100t** [comment]
how can i see if mesh service is running ?

same problem after update.sh --force

romain

**Reply by dinger1986** [reply]
Run the troubleshooting script or sudo systemctl status meshcentral

**Reply by info100t** [reply]
meshcentral.service - MeshCentral Server
     Loaded: loaded (/etc/systemd/system/meshcentral.service; enabled; vendor p>
     Active: active (running) since Sat 2023-02-04 20:22:47 CET; 4min 35s ago
   Main PID: 12331 (node)
      Tasks: 22 (limit: 4675)
     Memory: 483.2M
        CPU: 19.582s
     CGroup: /system.slice/meshcentral.service
             ├─12331 /usr/bin/node node_modules/meshcentral
             └─12397 /usr/bin/node /meshcentral/node_modules/meshcentral --laun>

févr. 04 20:22:47 rmm systemd[1]: Started MeshCentral Server.
févr. 04 20:22:48 rmm node[12331]: Installing mongodb@4.12.1...
févr. 04 20:23:17 rmm node[12331]: MeshCentral HTTP redirection server running >
févr. 04 20:23:17 rmm node[12331]: MeshCentral v1.1.3, WAN mode, Production mod>
févr. 04 20:23:18 rmm node[12331]: MeshCentral Intel(R) AMT server running on m>
févr. 04 20:23:18 rmm node[12331]: MeshCentral HTTP server running on port 4430>
févr. 04 20:23:18 rmm node[12331]: Loaded web certificate from "https://mesh.in>
févr. 04 20:23:18 rmm node[12331]:   SHA384 cert hash: 774a9f1b144f4923382c32e3>
févr. 04 20:23:18 rmm node[12331]:   SHA384 key hash: a473a4149d4c6c362e0

**Reply by dinger1986** [reply]
And if you got to your mesh domain is it working?

If so run the mesh troubleshooting again

**Reply by info100t** [reply]
yes, https://mesh.info100t.fr work :(

i run the script ;)

**Reply by dinger1986** [reply]
Please delete that asap as it has tokens in there.

**Reply by info100t** [reply]
it's done ;)

**Reply by dinger1986** [reply]
And regenerate them.
How to do that's in the docs.

I can't load mesh.yourdomain.com

**Reply by info100t** [reply]
So... error : Error getting mesh device id:
[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)

when i do : /rmm/api/env/bin/python /rmm/api/tacticalrmm/manage.py check_mesh

**Reply by info100t** [reply]
Now you can load mesh.domain.xx ? (my firewall block all countries but france).

**Reply by dinger1986** [reply]
Yeah I can.

Ok so not that. What's in your hosts file?

**Reply by info100t** [reply]
hosts file ?

**Reply by dinger1986** [reply]
`cat /etc/hosts`

**Reply by info100t** [reply]
tactical@rmm:~$ cat /etc/hosts
127.0.0.1       rmm.info100t.fr rmm apirmm.info100t.fr rmm.info100t.fr mesh.info100t.fr localhost
127.0.1.1       rmm.info100t.fr rmm apirmm.info100t.fr rmm.info100t.fr mesh.info100t.fr localhost
80.11.205.160   rmm.info100t.fr rmm apirmm.info100t.fr rmm.info100t.fr mesh.info100t.fr localhost

**Reply by dinger1986** [reply]
Can you share the output of `cat /meshcentral/meshcentral-data/config.json` please

**Reply by info100t** [reply]
{
  "settings": {
    "Cert": "mesh.info100t.fr",
    "MongoDb": "mongodb://127.0.0.1:27017",
    "MongoDbName": "meshcentral",
    "WANonly": true,
    "Minify": 1,
    "Port": 4430,
    "AliasPort": 443,
    "RedirPort": 800,
    "AllowLoginToken": true,
    "AllowFraming": true,
    "_AgentPing": 60,
    "AgentPong": 300,
    "AllowHighQualityDesktop": true,
    "TlsOffload": "127.0.0.1",
    "agentCoreDump": false,
    "Compression": true,
    "WsCompression": true,
    "AgentWsCompression": true,
    "MaxInvalidLogin": { "time": 5, "count": 5, "coolofftime": 30 }
  },
  "domains": {
    "": {
      "Title": "Tactical RMM",
      "Title2": "Tactical RMM",
      "NewAccounts": false,
      "CertUrl": "https://mesh.info100t.fr:443/",
      "GeoLocation": true,
      "CookieIpCheck": false,
      "mstsc": true
    }
  }
}

**Reply by dinger1986** [reply]
Looks fine.

Are agents still overdue?

**Reply by info100t** [reply]
just a new agent i try to install

**Reply by dinger1986** [reply]
Is your av maybe blocking it?

**Reply by info100t** [reply]
i think about this. i try installler on 2 computers and i have the same error.

**Reply by dinger1986** [reply]
Can you disable your AV? It's funny it's showing a cert error as doesn't seem to be an issue

**Reply by info100t** [reply]
i try on blanck virtual machine, same error ;(

**Reply by silversword411** [reply]
You change the settings in TRMM or mesh breaking the link?

https://docs.tacticalrmm.com/troubleshooting/#agents-not-installing-or-updating

**Reply by info100t** [reply]
What kind of settings ? i just change the cert to use my own.

Mesh site: https://mesh.info100t.fr
Mesh username: XXX
Mesh token: XXX
Mesh device group: XXX
Auth token ok:
XXX
Mesh url ok:
wss://mesh.info100t.fr/XXX
Error getting mesh device id:
[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)

**Reply by wh1te909** [reply]
problem is your cert. either not using a fullchain or don't have CA's installed.

**Reply by info100t** [reply]
fyi :

@rmm:~# tail -f /rmm/api/tacticalrmm/tacticalrmm/private/log/django_debug.log
    transport, protocol = await self._create_connection_transport(
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 1122, in _create_connection_transport
    await waiter
  File "/usr/local/lib/python3.10/asyncio/sslproto.py", line 534, in data_received
    ssldata, appdata = self._sslpipe.feed_ssldata(data)
  File "/usr/local/lib/python3.10/asyncio/sslproto.py", line 188, in feed_ssldata
    self._sslobj.do_handshake()
  File "/usr/local/lib/python3.10/ssl.py", line 975, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)

**Reply by wh1te909** [reply]
fix your cert. use a fullchain or fix your ca store on your server

**Reply by info100t** [reply]
how can i do that ?

**Reply by info100t** [reply]
i found !!!!!

my fullchain is break... i reconstruct my cert and now it ok !

thanks for all guys

**Reply by wh1te909** [reply]
awesome :)

**Reply by dinger1986** [reply]
Awesome!
